animation_nodes/data_structures/meshes/mesh_data.pyx

```diff
@@ -140,37 +150,51 @@ cdef class Mesh:
         if self.getAttributeDomainLength(attribute.domain) != len(attribute.data):
             raise Exception("invalid length")
 
-        self.attributes[attribute.name] = attribute
+        if attribute.type == AttributeType.MATERIAL_INDEX:
+            self.materialIndices = attribute
+        elif attribute.type == AttributeType.UV_MAP:
+            self.uvMapAttributes[attribute.name] = attribute
+        elif attribute.type == AttributeType.VERTEX_COLOR:
+            self.vertexColorAttributes[attribute.name] = attribute
+        elif attribute.type == AttributeType.MATERIAL_INDEX:
+            self.materialIndices = attribute
+        else:
+            self.customAttributes[attribute.name] = attribute
 
     def getAttributes(self, AttributeType type):
-        attributes = list()
-        for name, attribute in self.attributes.items():
-            if type == attribute.type: attributes.append((name, attribute))
-            elif type == AttributeType.VERTEX_COLOR and attribute.dataType == AttributeDataType.BYTE_COLOR:
-                attributes.append((name, attribute))
-        return attributes
+        if type == AttributeType.UV_MAP:
+            return list(self.uvMapAttributes.items())
+        elif type == AttributeType.VERTEX_COLOR:
+            return list(self.vertexColorAttributes.items())
+        else:
+            attributes = list()
+            for name, attribute in self.customAttributes.items():
+                if type == attribute.type: attributes.append((name, attribute))
+            return attributes
 
     def getAttributeNames(self, AttributeType type):
```

Comment:
This is almost identical to the above method. Just call the above method in a list comprehension and extract the names.

This also should be specialized. `getAllCustomAttributeNames`, `getAllVertexColorNames`, `getAllUVMapNames`.